src/screens/OtherProfile/index.js

```diff
@@ -177,11 +179,7 @@ const OtherProfile = () => {
   const onShare = async () => ShareUtils.shareUserLink(username);
 
   const handleSetUnFollow = async () => {
-    setDataMain((prevState) => ({
-      ...prevState,
-      is_following: false,
-      is_me_following_target: false
-    }));
+    setIsCurrentFollowed(false);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change is an improvement over the previous code as it simplifies the state management by only updating the relevant piece of state (`isCurrentFollowed`) instead of the entire `dataMain` object. However, please ensure that there are no other parts of your application that rely on the `is_following` or `is_me_following_target` properties of `dataMain`, as these are no longer being updated here.



<!-- This is an auto-generated comment by OSS CodeRabbit -->